docs/tutorial/batch-minting.md

```diff
@@ -0,0 +1,246 @@
+---
+ # Batch-minting NFTs on Ethereum
+
+ > TODO: Add missing links
+
+ This tutorial addresses some techniques in batch-minting big numbers of Ethereum non-fungible tokens (NFTs) as well as common idioms of uploading files and metadata on to IPFS/Filecoin via nft.storage.
+
+See also: [Lazy minting](./lazy-minting.md)
+
+Quite often,  NFT projects consist of hundreds if not thousands of files and metadata starting their lives out on a computer file system, waiting nervously to be part of a mint.
+
+There are many cases in which instead of minting NFTs just in time (or lazily) once the buyer has concluded a purchase, it is desirable to mint several all at once. Some common ones are:
+
+- Event tickets
+- Packs
+- Game items
+
+In all these scenarios, NFTs are being treated closer to commodities than in the case in which collectors collect them as discrete artworks. Some challenges that lie in minting a batch of many tokens include exceedingly high gas fees on the Ethereum network, atomically synchronizing between mint and upload failures, and performance of uploading files to nft.storage, etc.
+
+## Pre-processing and project structure
+
+> TODO: Add content here
+
+## Storage
+
+### Upload files iteratively
+
+The most straightforward way to upload files and metadata to nft.storage is by iterating over all of them and calling [`NFTStorage.store`]() one-by-one. Being simple as it is, that means you are responsible for properly handling errors that might occur for each successive request and synchronize it with the minting process.
+
+```javascript
+
+async function readAndUpload() {
+  const promises = Array.from({length: 10}, (v, i) => i)
+      .map(async (i) => {
+        const image = (await fetch(`$creature-{i}.jpg`)).blob()
+        return {
+          image,
+          name: `$creature-{i}`,
+          description: `Metadata of creature-${i}`,
+        }
+      })
+
+  const client = new NFTStorage({ token })
+  const results = await Promise.all(ps)
+
+  await Promise.all(results.forEach((result) => {
+    const metadata = await client.store(nft)
+    console.log('NFT data stored!')
+    console.log('Metadata URI: ', metadata.url)
+  }))
+}
+
+```
+
+### Upload directory of files
+
+A straightforward way is to employ the [NFTStorage.storeDirectory]() method to upload all the files stored in a local directory. This method handles many things under the hood for you, including rate-limiting for each request.
+
+```javascript
+
+import path from 'path'
+import { NFTStorage } from 'nft.storage'
+import { fileFromPath } from 'files-from-path'
+
+async function readAndUploadDir() {
+  const files = filesFromPath(dirpath, {
+		pathPrefix: path.resolve(dirpath),
+		hidden: true,
+	})
+  const storage = new NFTStorage({ token })
+  const cid = await storage.storeDirectory(files)
+  const status = await storage.status(cid)
+
+  console.log(status)
+}
+
+```
+
+## Smart contracts
+
```

Comment:
Add a preface / summary here, like "Now that you've uploaded all the assets, you can choose from various contract standards to mint the tokens."